src/App.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Header from './components/Header';
+import Editor from './components/Editor';
+import Canvas from './components/Canvas';
+import './styles/App.scss';
+import {TInfographicsSection} from './types/TInterfaceSection';
+
+interface IState {
+    infographicsSections: TInfographicsSection[],
+    numberOfAddedSections: number,
+    selectedSectionId: number
+}
+
+class App extends React.Component<{}, IState> {
+
+    state = {
+        infographicsSections: [] as TInfographicsSection[],
+        numberOfAddedSections: 0,
+        selectedSectionId: -1
+    };
+
+    public render() {
+
+        const {infographicsSections, selectedSectionId} = this.state;
+
+        return (
+            <div className="app">
+                <Header/>
+                <div className="app-wrapper">
+                    <Editor
+                        infographicsSections={infographicsSections}
+                        selectedSectionId={selectedSectionId}
+                        addSection={this.handleAddSection}
+                        deleteSection={this.handleDeleteSection}
+                    >
+                    </Editor>
+                    <Canvas
+                        infographicsSections={infographicsSections}
+                        selectedSectionId={selectedSectionId}
+                        handleSelectSection={this.handleSelectSection}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private handleAddSection = () => {
+
+        const {infographicsSections, numberOfAddedSections, selectedSectionId} = this.state;
+
+        const newSection: TInfographicsSection = {
+            id: numberOfAddedSections,
+            isActive: false
+        };
+
+        this.setState({
+            infographicsSections: [...infographicsSections, newSection],
+            numberOfAddedSections: numberOfAddedSections + 1,
+            selectedSectionId: selectedSectionId
+        });
+    };
+
+    private handleSelectSection = (id: number) => {
+
+        const {infographicsSections, numberOfAddedSections} = this.state;
+
+        this.setState({
+            infographicsSections: infographicsSections,
+            numberOfAddedSections: numberOfAddedSections,
```

Comment:
do setState nemusis davat puvodni hodnoty co se nemeni, v tomto pripade menis jen ID 

staci tedy

`this.setState({selectedSectionId})` a zmeni se ti jen ID, zbytek se nezmeni

parametr ID bych prejmenoval na selectedSectionId, at muzes pouzit shorthand v setState